anonymizer_module/metrics_statistics/central_server_client.py

```diff
@@ -0,0 +1,122 @@
+#  The MIT License
+#  Copyright (c) 2021- Nordic Institute for Interoperability Solutions (NIIS)
+#  Copyright (c) 2017-2020 Estonian Information System Authority (RIA)
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+import re
+import xml.etree.ElementTree as ET
+from logging import Logger
+from typing import Dict, Sequence, TypedDict
+
+import requests
+
+
+class MemberCountData(TypedDict):
+    member_gov_count: int
+    member_com_count: int
+    member_org_count: int
```

Comment:
The number and type of member classes isn't fixed and it can vary between different X-Road ecosystems. The member classes are defined on the Central Server and the X-Road Operator is free to decide how many classes there are and what are the types. For example, these are the member classes of the Estonian and Finnish X-Road ecosystems:

**Estonia**
* COM
* GOV
* NEE
* NGO

**Finland**
* COM
* EDU
* GOV
* MUN
* ORG

Would it be possible to make the member classes configurable?

Reply:
> The number and type of member classes isn't fixed and it can vary between different X-Road ecosystems. The member classes are defined on the Central Server and the X-Road Operator is free to decide how many classes there are and what are the types. For example, these are the member classes of the Estonian and Finnish X-Road ecosystems:
> 
> **Estonia**
> 
>     * COM
> 
>     * GOV
> 
>     * NEE
> 
>     * NGO
> 
> 
> **Finland**
> 
>     * COM
> 
>     * EDU
> 
>     * GOV
> 
>     * MUN
> 
>     * ORG
> 
> 
> Would it be possible to make the member classes configurable?

I think rather than having them be configurable, we could just use the ones that are parsed out of the global configuration so that the information is always based on the actual instance. Would that be ok in your view as well @petkivim?

Reply:
Thanks for comment. I could treat member class counts  as dynamic and strore it in PG json field as with service counts. Does it sound reasonable? I could take classes from global configuration and count it occurrences in memeber list.

Reply:
Treating member classes and their member counts dynamically sounds good to me.